zygrader/bobs_shake.py

```diff
@@ -0,0 +1,422 @@
+"""Bob's Shake: A tool to analyze TA work statistics.
+
+`shake` is the singular entry point to this module
+"""
+# TODO: Check this against the old Bob's Second
+
+# TODO: run my 'history fixer' on the lock log
+# (explained in a FIXME below, which should also be removed)
+
+from collections import namedtuple
+import csv
+import datetime
+import os
+import typing
+
+from zygrader import data, ui
+from zygrader.data.lock import get_lock_log_path
+from zygrader.config import preferences
+from zygrader.ui.templates import filename_input
+
+
+def shake():
+    """Read TA work data, "shake" it, and report the shaken statistics.
+
+    Results summarize work in grading, answering emails, and helping students.
+    For each category of work, Bob's Shake reports the number of items finished
+    by a TA and the total time spent on those items.
+
+    Resulting summary statistics are written to a file chosen by the user.
+
+    Some TA data is read from zygrader's own native lock log,
+    and additional data is read from a csv file created by our help queue.
+
+    "shaking" occurs in a series of steps, roughly:
+        - user selects start and end time to include
+        - user points Bob to the data from the help queue
+        - data is read from zygrader's log and the queue info
+        - errors encountered and corrections that need to be made
+          are presented to the user
+        - input data is assigned to each TA appropriately
+        - each TA's events are analyzed to calculate the results summary
+        - user selects an output file
+        - the summary statistics are written to the file
+    """
+    window = ui.get_window()
+    wait_popup = ui.layers.WaitPopup("Bob's Shake")
+
+    worker = _StatsWorker()
+    Step = namedtuple('Step', ['interactive', 'msg', 'func'])
+    steps = [
+        Step(True, "start time", lambda: worker.select_start_time()),
+        Step(True, "end time", lambda: worker.select_end_time()),
+        Step(True, "select helpqueue csv",
```

Comment:
Probably use title case or at least capitalize the messages here

Reply:
for the interactive steps, the msg is never presented, so I was purposefully leaving these lower case. they only really serve to help in reading the code, and I'm realizing the func name does this well enough, so I'll actually just make them empty I think